src/components/List/List.tsx

```diff
@@ -1,168 +1,80 @@
+    filter: string;
+    activeItem?: number;
+    pageSize?: number;
+    sorting?: boolean;
+};
+
+export const listDefaultProps: Partial<ListProps<ListItemData<unknown>>> = {
     items: [],
     itemClassName: '',
-
     filterable: true,
     sortable: false,
     virtualized: true,
-
     deactivateOnLeave: true,
 };
 
-export class List extends React.Component {
-    static moveListElement(list, oldIndex, newIndex) {
+export class List<T = unknown> extends React.Component<ListProps<T>, ListState<T>> {
+    static defaultProps: Partial<ListProps<ListItemData<unknown>>> = listDefaultProps;
+
+    static moveListElement(list: ListItemData<unknown>[], oldIndex: number, newIndex: number) {
         if (oldIndex !== newIndex) {
             const [item] = list.splice(oldIndex, 1);
             list.splice(newIndex, 0, item);
         }
+
         return list;
     }
-    static findNextIndex(list, index, step) {
+
+    static findNextIndex(list: ListItemData<unknown>[], index: number, step: number) {
         const dataLength = list.length;
         let currentIndex = (index + dataLength) % dataLength;
+
         for (let i = 0; i < dataLength; i += 1) {
             if (list[currentIndex] && !list[currentIndex].disabled) {
                 return currentIndex;
             }
             currentIndex = (currentIndex + dataLength + step) % dataLength;
         }
+
         return undefined;
     }
-    static propTypes = {
-        items: PropTypes.array.isRequired,
-        itemHeight: PropTypes.oneOfType([PropTypes.number, PropTypes.func]),
-        itemsHeight: PropTypes.oneOfType([PropTypes.number, PropTypes.func]),
-        itemClassName: PropTypes.string,
-
-        theme: PropTypes.string,
-        view: PropTypes.string,
-        tone: PropTypes.string,
-        size: PropTypes.string,
-        itemsClassName: PropTypes.string,
-        className: PropTypes.string,
-
-        renderItem: PropTypes.func,
-        filterItem: PropTypes.func,
-        onFilterEnd: PropTypes.func,
-
-        filterable: PropTypes.bool,
-        filter: PropTypes.string,
-        filterPlaceholder: PropTypes.string,
-        emptyPlaceholder: PropTypes.string,
-        filterClassName: PropTypes.string,
-        onFilterUpdate: PropTypes.func,
-
-        virtualized: PropTypes.bool,
-        sortable: PropTypes.bool,
-        sortHandleAlign: PropTypes.oneOf(['left', 'right']),
-        onSortEnd: PropTypes.func,
-
-        deactivateOnLeave: PropTypes.bool,
-        activeItemIndex: PropTypes.number,
-        selectedItemIndex: PropTypes.number,
-
-        onItemClick: PropTypes.func,
-    };
-    static defaultProps = listDefaultProps;
 
-    state = {
+    state: ListState<T> = {
         items: this.props.items,
         filter: '',
     };
-    componentDidUpdate(prevProps) {
+
+    refFilter = React.createRef<HTMLInputElement | HTMLTextAreaElement>();
+    refContainer = React.createRef<any>();
+    blurTimer: NodeJS.Timeout | null = null;
```

Comment:
NodeJS??

Reply:
Because `setTimeout` has this type:
```typescript
declare function setTimeout(callback: (...args: any[]) => void, ms?: number, ...args: any[]): NodeJS.Timeout;
```

Reply:
I think that it is an error, but typescript have another opinion

Reply:
Where this type comes from? Which file?